master/src/main/java/org/apache/celeborn/service/deploy/master/MasterSecretRegistryImpl.java

```diff
@@ -44,7 +44,38 @@ public void register(String appId, String secret) {
     }
   }
 
-  void setMetadataHandler(IMetadataHandler metadataHandler) {
+  @Override
+  public String getSecretKey(String appId) {
+    String secret = super.getSecretKey(appId);
+    if (secret == null && metadataHandler != null) {
+      LOG.info("Fetching secret from metadata manager for appId: {}", appId);
+      ApplicationMeta applicationMeta = metadataHandler.applicationMetas.get(appId);
+      if (applicationMeta != null) {
+        secret = applicationMeta.secret();
+        LOG.debug("Found Secret from metadata manager for appId: {}, register secret", appId);
```

Comment:
We could make this an `info` and make the `info` above (for `"Fetching secret from metadata manager for appId"`) to debug ?

Same for `isRegistered` below